spec/spec_helper_acceptance.rb

```diff
@@ -130,11 +130,10 @@ def retry_on_error_matching(max_retry_count = 3, retry_wait_interval_secs = 5, e
         end
 
         if fact_on(host, 'osfamily') == 'windows'
-          win_host = only_host_with_role(hosts, 'default')
-          @windows_ip = win_host.ip
-          apply_manifest_on(host, "class { 'docker': docker_ee => true, extra_parameters => '\"insecure-registries\": [ \"#{@windows_ip}:5000\" ]' }")
+          apply_manifest_on(host, "class { 'docker': docker_ee => true }")
           docker_path = "/cygdrive/c/Program Files/Docker"
           host.add_env_var('PATH', docker_path)
+          host.add_env_var('TEMP', 'C:\Users\Administrator\AppData\Local\Temp')
```

Comment:
This still seems very wrong.  The host should already have a correctly setup TEMP directory.  

Even though the cygwin environment says `\tmp`, external processes created from within cygwin use `C:\cygwin64\tmp`

Windows 2012R2 Pooler VM

``` text
Administrator@adx7v3xdcx8nwfn ~
$ export
...
declare -x TEMP="/tmp"
declare -x TERM="xterm-256color"
declare -x TMP="/tmp"
...

Administrator@adx7v3xdcx8nwfn ~
$ cmd.exe /c set temp
TEMP=C:\cygwin64\tmp

Administrator@adx7v3xdcx8nwfn ~
$ powershell.exe -Command "Write-Host \$ENV:TEMP"
C:\cygwin64\tmp
```

Reply:
That is the way it should work. Even though facter shows the fact pointing to `C:\cygwin64\tmp`,
when running tests, the fact does not seem to be initialized:
`Warning: Unknown variable: '::docker_user_temp_path'`
This happens if I don't set the TEMP env var from within the spec_helper_acceptance...
This works fine for the other 2 facts that are being read from the environment.

Reply:
It's difficult to understand how overriding a system set env variable is the right way to go, especially since it's being hard coded to a specific user's temp path that may or may not be the correct user for a particular run.

Once this variable is overwritten in this way, it's hard to tell what might be broken down the road. Doesn't it seem like this issue you are seeing might be down to misbehavior on the part of beaker or cygwin, that should be worked around more formally?

Reply:
Unfortunately, I have not found a way to work around this.
After doing some more digging around, it looks like the user profile is not being loaded when beaker runs the commands.
The commands are being run remote through ssh: `ssh Administrator@host -i key cmd /c puppet apply /tmp/manifest`

Logging in with ssh will load your user profile and give you the TMP env var but running it remotely will not:

```
ssh Administrator@e34ei8s1mokqrsl.delivery.puppetlabs.net -i .ssh/id_rsa-acceptance
-bash-4.3$ cmd /c set TMP
TMP=C:\cygwin64\tmp
```
vs
```
ssh Administrator@e34ei8s1mokqrsl.delivery.puppetlabs.net -i .ssh/id_rsa-acceptance cmd /c set TMP
Environment variable TMP not defined
```

Adding the variable in `.ssh/environment` would probably fix this, but we still need to enter the value manually.

I agree, it's not right to hardcode the username but since we are running on the vmpooler VMs, we can assume it is running under Administrator.